core/state/statedb.go

```diff
@@ -101,6 +101,8 @@ type StateDB struct {
 	stateObjectsPending map[common.Address]struct{} // State objects finalized but not yet written to the trie
 	stateObjectsDirty   map[common.Address]struct{} // State objects modified in the current execution
 
+	storagePool  *StoragePool // shared_pool to store L1 originStorage of stateObjects
+	dbForSpeedup bool
```

Comment:
What about  `writeOnSharedStorage bool`, with the comment: `Write to the shared origin storage of a stateObject while reading from the underlying storage layer.`